### ISSUE TITLE
Hourly Trend Widget 24 Hours Like AF Weather

### ISSUE THREAD

**Issue by WithThisHerring** [breezy-weather/breezy-weather#252]
### Describe your idea

AF Weather is a great 24-hour trend weather widget:
https://github.com/Gitsaibot/AF-Weather-Widget

It has an excellent layout and shows a lot of information compactly.  Temperature, rain maximum and minimum, weather icons.

The current hourly trend widget just shows 5 hour marks.  Usually, one knows by time of day whether the temperature is going up or down in just the next few hours.  A 24-hour graph widget would give a much better perspective.

It seems silly, but I currently have two weather apps on my phone (Breezy and AF) to get a nice set of widgets on my home screen.  AF is a great summary, and I use Breezy for all the detail.  I would love if Breezy could do it all.

(Thank you!  This app fixes a lot of the Geometric issues I've had :)  )

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] My idea is not related to an existing feature described in the [Homepage](https://github.com/breezy-weather/breezy-weather/blob/main/docs/HOMEPAGE.md) document.
- [X] I have updated the app to **[latest version](https://github.com/breezy-weather/breezy-weather/releases/latest)**.

**Comment by papjul** [comment]
Yes, definitely something I want to work on, with a priority on #14 as these ones are really not easy to see.

I didn't have widgets in mind, but I think that's a good idea.

**Comment by Otiel** [comment]
+1 for a widget showing the next 24-hours. I would even prefer to have the next 48-hours: in that case, at any time of the day, the current day and the next day is fully displayed on the widget.

I currently use eWeather HDF ([app store link](https://play.google.com/store/apps/details?id=com.Elecont.WeatherClock.free)) for its widget in combination with Breezy. With a little customization, this is the widget I managed to have:

![image](https://github.com/breezy-weather/breezy-weather/assets/3851902/7cbb5b2d-fe85-4370-8394-d55b0bed89de)

It shows 16 marks with weather info, each one representing a 3-hour window (16*3=48 hours total). For each 3-hour window, the following info is displayed (top to bottom):
* Temperature
* Wind direction (represented by an arrow)
* Wind speed (green)
* Precipitation probability (blue)
* Icon representing the weather
* Precipitation amount (blue bars)
* Start time of the 3-hours window, with the first one being the last refresh time.

It is a lot of information, font is rather small (but not smaller than hour / battery percentage) in order to fit everything on the screen, but it's honestly the only weather widget among all those I've tried that is actually useful. The only drawback I currently have with this widget is that weather data is not based on the source I wish to use (Météo France). This is why I'd love for Breezy to provide such a widget. ❤️ 

Full capture for context:

![Screenshot_20230822_152900_Nova7](https://github.com/breezy-weather/breezy-weather/assets/3851902/d2be9e6e-c280-48b1-b3fa-e7fb706c92a2)

**Comment by spitfire94** [comment]
This is a much-needed enhancement to the daily and hourly trends widgets. Would make this perfect app all the more better!

**Comment by apraile** [comment]
It would be great to have this feature.
As a suggestion, there is another 24-hour widgets that I like and it is the one used by [Cirrus](https://github.com/woheller69/omweather). Emulates an analog clock:

![Screenshot_20240820-110208_1 cleaned](https://github.com/user-attachments/assets/dbc66ff7-81a5-4d9e-875e-fb3d0ea646f0)

Thanks to the developers for the app :heart:

**Comment by 0-5-0** [comment]
Could a similar view be added to the normal app? It feels bad having to press the various tabs under the hourly or daily views to see Temperature, Precipitation, UV, Wind, etc.. It would definitely be nice to be able to see various data sets in one chart, but not necessarily exclusive only to homescreen widgets.

**Comment by papjul** [comment]
The 24-hour window-fitting chart is planned as part of #14, however there is no plan to mix different types of data on a single chart, neither in app or widgets (see #939 for more info).

If you want this feature, you will need to do this on your own app through the `WeatherSpec` receiver we expose, or through the future `ContentProvider` #863 if you can wait.
